domain/credential/state/state.go

```diff
@@ -386,23 +386,62 @@ AND    cloud_credential.cloud_uuid = (
 	return errors.Trace(err)
 }
 
-// CloudCredentialsForOwner returns the owner's cloud credentials for a given cloud,
-// keyed by credential name.
+// CloudCredentialsForOwner returns the owner's cloud credentials for a given
+// cloud, keyed by credential name.
 func (st *State) CloudCredentialsForOwner(ctx context.Context, owner user.Name, cloudName string) (map[string]credential.CloudCredentialResult, error) {
 	db, err := st.DB()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	var creds []credential.CloudCredentialResult
+	var (
+		dbRows      Credentials
+		dbAuthTypes []authType
+		keyValues   []CredentialAttribute
+	)
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		var err error
-		creds, err = LoadCloudCredentials(ctx, st, tx, "", cloudName, owner)
```

Comment:
LoadCloudCredentials was overloaded to have 3 different functions based on the inputs. It relied on the (now deleted) `SqlairAndClause` to add conditions to the WHERE clause if the arguments were none zero.

I have split it into different functions to capture the different use cases, hopefully making it clear to readers of the code what is going on.